web/src/stream_topic_history.ts

```diff
@@ -63,16 +70,29 @@ export function is_complete_for_stream_id(stream_id) {
     return in_cache;
 }
 
-export function stream_has_topics(stream_id) {
+export function stream_has_topics(stream_id: number): boolean {
     if (!stream_dict.has(stream_id)) {
         return false;
     }
 
     const history = stream_dict.get(stream_id);
+    assert(history !== undefined);
 
     return history.has_topics();
 }
 
+export type Topic = {
+    historical: boolean;
+    count: number;
+    message_id: number;
+    pretty_name: string;
+};
```

Comment:
from [here](https://github.com/zulip/zulip/pull/28414/commits/487e58b8bb09f13339aad7520e258d574e59091c#r1440884844

> `TopicHistoryEntry` might be a name for this? I don't love the idea of calling it a `Topic`, since it doesn't contain topic-settings type fields.

Reply:
Yeah sure! I'll add `Topic` type back to `unread.ts` as something separate then

Reply:
Oh actually `unread.get_missing_topics` needs something of the same type, so they're connected. I'll go with `unread` importing `TopicHistoryEntry`, let me know if that doesn't make sense to you